src/server/verification/verificationAPI.js

```diff
@@ -112,7 +112,21 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
             }
           }
         } else {
+          //only approved users can do the process
+          const isApprovedToClaim = user.claimQueue && ['approved', 'whitelisted'].includes(user.claimQueue.status)
+          if (conf.claimQueueAllowed > 0 && isApprovedToClaim === false)
+            throw new Error('User not approved to claim, not in queue or still pending')
           enrollmentResult = await enrollmentProcessor.enroll(user, enrollmentIdentifier, payload, log)
+          //if user has passed, then we mark that in claim queue and tag the user
+          if (enrollmentResult.isVerified) {
+            await Promise.all([
+              user.claimQueue &&
+                storage.updateUser({ identifier: user.identifier, 'claimQueue.status': 'whitelisted' }),
+              Mautic.addContactsToSegment([user.mauticId], conf.mauticClaimQueueWhitelistedSegmentId).catch(e => {
```

Comment:
Why we need to integrate with Mautic ? users awaiting in the claim queue could receive some specific email (e.g you'll be available to claim in xx days approximately) ?

I think it would be better to consolidate all logic about claim queue (incl Mautic integration) in the single serivce. Not spreading it through all server code

Reply:
yes thats why we integrate with mautic
ok i'll check if it makes sense to consolidate code

Reply:
Thanks!